src/main/template/presto.yaml

```diff
@@ -0,0 +1,652 @@
+                #!/bin/bash -xe
+                mkdir /opt/presto
+                wget -q -O - "https://repo1.maven.org/maven2/io/prestosql/presto-server/326/presto-server-326.tar.gz" | tar xvzf - -C /opt/presto
+                mv "/opt/presto/presto-server-326" /opt/presto/presto-server
+                mkdir /var/presto
+                mkdir /var/presto/data
+        configure_presto:
+          files:
+            /opt/presto/presto-server/etc/node.properties:
+              content: |
+                node.data-dir=/var/presto/data
+                node.environment={{nodeEnvironment}}
+              context:
+                nodeEnvironment: !Ref environmentName
+              owner: root
+              group: root
+              mode: "000644"
+            /opt/presto/presto-server/etc/jvm.config:
+              content: |
+                -server
+                -XX:-UseBiasedLocking
+                -XX:+UseG1GC
+                -XX:G1HeapRegionSize=32M
+                -XX:+ExplicitGCInvokesConcurrent
+                -XX:OnOutOfMemoryError="kill -9 %p"
+                -XX:+UseGCOverheadLimit
+                -XX:ReservedCodeCacheSize=512M
+                -Djdk.attach.allowAttachSelf=true
+                -Djdk.nio.maxCachedBufferSize=2000000
+              owner: root
+              group: root
+              mode: "000644"
+            /opt/presto/presto-server/etc/config.properties:
+              content: |
+                http-server.http.port=8080
+                query.max-memory=50GB
+                query.max-memory-per-node=1GB
+                query.max-total-memory-per-node=2GB
+                coordinator=true
+                node-scheduler.include-coordinator=false
+                discovery-server.enabled=true
+                discovery.uri=http://{{discoveryUri}}:8080
+              context:
+                discoveryUri:
+                  Fn::Join:
+                    - ""
+                    - - "coordinator."
+                      - !Ref internalDomain
+              owner: root
+              group: root
+              mode: "000644"
+            /opt/presto/presto-server/etc/catalog/cassandra.properties:
+              content: |
+                connector.name=cassandra
+                cassandra.contact-points={{cassandraContactPoints}}
+                cassandra.consistency-level=LOCAL_QUORUM
+                cassandra.client.read-timeout=2m
+              context:
+                cassandraContactPoints:
+                  Fn::ImportValue:
+                    !Sub "${cassandraStackName}-seed-nodes"
+              owner: root
+              group: root
+              mode: "000644"
+          commands:
+            patch_jvm_config:
+              command: |
+                export SYSTEM_MEMORY_IN_MB=`free -m | awk '/:/ {print $2;exit}'`
+                export HEAP_SIZE=`expr $SYSTEM_MEMORY_IN_MB / 4 \* 3`
+                echo "-Xms${HEAP_SIZE}M" >> /opt/presto/presto-server/etc/jvm.config
```

Comment:
Setting minimum heap and maximum heap to the same value prevents the JVM from finding a heap size that is optimal for its current load in terms of garbage collection (full GC on large heap obviously is longer than on a smaller one).

Let's do one of the following options:

- Perform a multi-hour benchmark to see whether this fixed-size heap handles typical load better, same or worse than a dynamically sized one. If it's significantly better, state that in a comment in the yaml file and keep the `Xms` as is. Otherwise, go with one of the other options.
- Leave out the `Xms` setting so that JVM defaults apply (my preferred option).
- Set `Xms` to a small fixed value (e.g. 512M assuming `HEAP_SIZE` will never be below that anyway).

Reply:
... and then I noticed that `Xms` is already gone these days, having been removed as part of 77d220a03361012f25c4f7f5c051e83a432bfd6b.